src/block.rs

```diff
@@ -0,0 +1,314 @@
+use std::mem::transmute;
+
+use crate::crypto::{hash, PublicKey};
+use crate::time::create_timestamp;
+use crate::transaction::Transaction;
+
+/// The `Block` holds all data inside the block body,
+/// and additional metadata not to be serialized
+#[derive(PartialEq, Debug, Clone)]
+pub struct Block {
+    /// The body and content of the block object
+    pub body: BlockBody,
+    /// Byte array hash of the block
+    bsh: [u8; 32],
+}
+/// This `BlockBody` holds data to be serialized along with
+/// `Transaction`s
+#[derive(PartialEq, Debug, Clone)]
+pub struct BlockBody {
+    /// Block id
+    pub id: u64,
+    /// Block timestamp
+    pub ts: u64,
+    /// Byte array hash of the previous block in the chain
+    pub previous_block_hash: [u8; 32],
+    /// `Publickey` of the block creator
+    pub creator: PublicKey,
+    /// List of transactions in the block
+    pub txs: Vec<Transaction>,
+    /// `BurnFee` containing the fees paid to produce the block
+    pub burnfee: u64,
+    /// Block difficulty required to win the `LotteryGame` in golden ticket generation
+    difficulty: f32,
+    /// Ratio determing the block reward split between nodes and miners
+    paysplit: f32,
+    /// Vote determining change of difficulty and paysplit
+    vote: i8,
```

Comment:
I believe this is what we've been calling "advanced saito" and shouldn't be in the first release.